cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/types/TypeParser.java

```diff
@@ -196,15 +162,8 @@ public static boolean isStorageSpecifier(String specifier) {
   }
 
   protected static boolean isQualifierSpecifier(String qualifier) {
-    // TODO: Convert to language trait
-    if (getLanguage() instanceof JavaLanguage) {
-      return qualifier.equals(FINAL_QUALIFIER) || qualifier.equals(VOLATILE_QUALIFIER);
-    } else {
-      return qualifier.equals(CONST_QUALIFIER)
-          || qualifier.equals(VOLATILE_QUALIFIER)
-          || qualifier.equals(RESTRICT_QUALIFIER)
-          || qualifier.equals(ATOMIC_QUALIFIER);
-    }
+    return getLanguage() instanceof HasQualifier
```

Comment:
Can we add language as an additional parameter here? I want to get rid of the `getLanguage` calls in the singleton type manager.